### PR TITLE
Switch from Travis to GitHub Actions 

### PR DESCRIPTION
**Pull request by jason-fox**
see telefonicaid/iotagent-ul#452

Docker file has been updated so Hadolint doesn't complain

### CODE REVIEW

**Comment by jason-fox** [comment]
Sample CI run: https://github.com/jason-fox/perseo-core/actions/runs/391924326

**Comment by IvanHdzC** [comment]
@jason-fox according to some other PRs on the platform's repossitories. Is it possible that it's missing a build webhook with some other labels?

**Comment by jason-fox** [comment]
@IvanHdzC - I'm not entirely sure what you mean.  The deleted travis file is just running test (no text lint, no checkstyle etc.) This PR replicates that but does not extend it. 

If you want to run a  [Matrix](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) of actions over multiple JDKs you could do that if you wish.

I think the idea of linting the codebase as part of the push would also be a good addition, but  it is not what I'm trying to achieve with this fly-by. Just saving Travis minutes.

**Comment by jason-fox** [comment]
Properly sorting out the `Dockerfile` with a multistage build and injecting more metadata labels, distroless option, non-root user etc. should be added as an issue/technical debt. This PR is purely CI not seriously updating Docker.

**Comment by IvanHdzC** [comment]
> @IvanHdzC - I'm not entirely sure what you mean. The deleted travis file is just running test (no text lint, no checkstyle etc.) This PR replicates that but does not extend it.
> 
> If you want to run a [Matrix](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) of actions over multiple JDKs you could do that if you wish.
> 
> I think the idea of linting the codebase as part of the push would also be a good addition, but it is not what I'm trying to achieve with this fly-by. Just saving Travis minutes.

I didn't explain myself correctly, sorry about that. Taking perseo-fe as example, there are a couple of [labels specified on a build hook ](https://github.com/jason-fox/perseo-fe/blob/master/docker/hooks/build) to include on the final image. 

- label"org.opencontainers.image.created"=$(date +%Y-%m-%d) \
- label "org.opencontainers.image.revision"=$SOURCE_COMMIT \
- label "org.opencontainers.image.version"=$DOCKER_TAG \

That's what I meant. Anyways, as you said, that's not the goal for this PR, so there is nothing to change.

NTC

**Comment by IvanHdzC** [comment]
LGTM

**Review by IvanHdzC — APPROVED** [review]
LGTM

**Comment by jason-fox** [comment]
Got it. The Dockerfile currently lies in root and overuses uses COPY/ADD hence adding `docker/hooks/build` would mean rewriting and retesting the whole Docker CI so best to add to technical debt for now so it doesn't get forgotten.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  unit-test:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v2
+      - name: Use Java 8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
```
Comment (fgalan) [inline .github/workflows/ci.yml:32]: In travis we have:

```
jdk:
  - openjdk8
  - oraclejdk8
  - oraclejdk9
  - openjdk10
  - openjdk11
```

so I understand we should include here also versions 9, 10 and 11 (maybe using `matrix` as I've seen in PRs for nodejs repos to cover node 10 and 12?)

Comment (jason-fox) [reply .github/workflows/ci.yml:32]: Yes, you should introduce a matrix build when you are ready.

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**